ratis-server-api/src/main/java/org/apache/ratis/statemachine/StateMachine.java

```diff
@@ -244,6 +244,14 @@ default void notifyConfigurationChanged(long term, long index, RaftConfiguration
      */
     default void notifyGroupRemove() {}
 
+    /**
+     * Notify the {@link StateMachine} a group removal event.
+     * @param applyTransactionFailure true if there was a failure in applying transaction.
+     */
+    default void notifyGroupRemove(boolean applyTransactionFailure) {
+      notifyGroupRemove();
+    }
```

Comment:
Please don't add `notifyGroupRemove(boolean applyTransactionFailure)` here.  Instead, it should fail the groupRemove request.

BTW, let's change groupRemove in a separated JIRA instead of changing it here.

Reply:
Will create a jira for this. Removed it from here.